src/main/java/Tasks.java

```diff
@@ -0,0 +1,173 @@
+            //System.err.println("Error loading tasks: " + e.getMessage());
+            tasks = new ArrayList<>();
+        }
+    }
+
+    public static void saveTask()  {
+        try (ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream(FILE_PATH))) {
+            oos.writeObject(tasks);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void markTask(String command) throws DukeException {
+        String[] words = command.split(" ");
+        if (words.length == 2) {
+            int taskId = Integer.parseInt(words[1]) -1;
+            if(taskId < tasks.size()) {
+                Task t = tasks.get(taskId);
+                t.markAsDone();
+                System.out.println(" Nice ! I've marked this task as done: \n" + t);
+            } else {
+                throw new DukeException(" Invalid task index. Type 'list' to list out your tasks. ");
+            }
+        } else {
+            throw new DukeException(" Invalid format. Type 'mark 1' to mark task 1. ");
+        }
+    }
+
+    public  void unmarkTask(String command) throws DukeException{
+        String[] words = command.split(" ");
+        if (words.length == 2) {
+            int taskId = Integer.parseInt(words[1]) - 1;
+            if (taskId < tasks.size()) {
+                Task t = tasks.get(taskId);
+                t.markAsUndone();
+                System.out.println(" Ok, I've marked this task as not done yet: \n" + t);
+            } else {
+                throw new DukeException(" Invalid task index. Type 'list' to list out your tasks. ");
+            }
+        } else {
+            throw new DukeException(" Invalid format. Use 'unmark 1' to unmark task 1. ");
+        }
+    }
+
+    public void addTask(String command) throws DukeException {
+        Task t;
+        switch (getTaskType(command)) {
+            case "todo":
+                t = handleTodo(command);
+                break;
+            case "deadline":
+                t = handleDeadline(command);
+                break;
+            case "event":
+                t = handleEvent(command);
+                break;
+            default:
+                throw new DukeException(" Sorry! I don't know what that means :c ");
+        }
+        tasks.add(t);
+        System.out.println(" Got it. I've added this task: \n" + t);
+        printTaskSize();
+    }
+
+    private void printTaskSize(){
+        System.out.println(" Now you have " + tasks.size() + " tasks in the list. ");
+
+    }
+
+    public void listOutTasks() {
+        if (tasks.isEmpty()) {
+            System.out.println(" No task added yet!");
+        } else {
+            System.out.println(" Here are the tasks in your list: ");
+            for(int i = 0; i < tasks.size(); i++) {
+                int seq = i+1;
+                Task t = tasks.get(i);
+                System.out.println(" " + seq + ". " + t.toString());
+            }
+        }
```

Comment:
Coding quality, make the happy path prominent. 

No need for the `else` for `!command.contains(" ")`

You can do like a guard clause style for easier readability and avoid complicated nesting.  
  
```java
if (tasks.isEmpty()) {
    // fail check
    System.out.println();
    return;
}
// rest of your code
...
```

The `return` will ensure the remaining code will not be executed anyways so no need for else. 

https://nus-cs2103-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#make-the-happy-path-prominent